src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPane.java

```diff
@@ -415,8 +415,17 @@ private void executeFunctionForObject(String function, String objectName)
    private Widget createImportMenu()
    {
       ToolbarPopupMenu menu = new ToolbarPopupMenu();
-      menu.addItem(commands_.importDatasetFromFile().createMenuItem(false));
-      menu.addItem(commands_.importDatasetFromURL().createMenuItem(false));
+      
+      if (prefs_.useDataImport().getValue())
+      {
+         menu.addItem(commands_.importDatasetFromCSV().createMenuItem(false));
```

Comment:
Right, one better than setting their visibility we can also remove them entirely. I'd have logic just like this but put it here instead and call .remove() on command that's aren't going to be used: https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/application/Application.java#L659

Note that uiPrefs_ is a member variable accessible from within `Application.initializeWorkbench()`